### PR TITLE
Add docs session to nox configuration for BigQuery

### PR DESCRIPTION
**Pull request by tswast**
* Run with `nox -s docs`. This allows us to just build the BigQuery docs
  to preview changes more quickly.

* Moves the magics module documentation to the top level. I found it
  impossible to debug the autosummary errors with generating the docs
  for the magics module, so I move those docs to our more standard
  automodule pattern. With automodule, the error messages were more
  clear (the IPython package was missing in my original implementation).
  Since this module is referenced in documentation and elsewhere, add a
  redirect to the new magics module documentation from the old generated
  location.

### CODE REVIEW

**Comment by tswast** [comment]
Test failure appears to be a flake. All sessions passed except `system-2.7`

```
Running session system-2.7
Creating virtualenv using python2.7 in /tmpfs/src/github/google-cloud-python/bigquery/.nox/system-2-7
pip install --upgrade --pre grpcio
pip install --upgrade mock pytest
pip install --upgrade -e ../api_core[grpc]
pip install --upgrade -e ../core
pip install --upgrade -e ../bigquery_storage[pandas,fastavro]
pip install --upgrade -e ../storage
pip install --upgrade -e ../test_utils
pip install --upgrade -e .[pandas]
pip install --upgrade ipython==5.5
py.test --quiet tests/system.py
......................F...........................                       [100%]
=================================== FAILURES ===================================
_______________________ TestBigQuery.test_list_datasets ________________________
self = <tests.system.TestBigQuery testMethod=test_list_datasets>
    def test_list_datasets(self):
        datasets_to_create = [
            "new" + unique_resource_id(),
            "newer" + unique_resource_id(),
            "newest" + unique_resource_id(),
        ]
        for dataset_id in datasets_to_create:
            self.temp_dataset(dataset_id)
        # Retrieve the datasets.
        iterator = Config.CLIENT.list_datasets()
        all_datasets = list(iterator)
        self.assertIsNone(iterator.next_page_token)
        created = [
            dataset
            for dataset in all_datasets
            if dataset.dataset_id in datasets_to_create
            and dataset.project == Config.CLIENT.project
        ]
>       self.assertEqual(len(created), len(datasets_to_create))
E       AssertionError: 2 != 3
tests/system.py:276: AssertionError
=============================== warnings summary ===============================
.nox/system-2-7/lib/python2.7/site-packages/IPython/external/decorators/__init__.py:2
  /tmpfs/src/github/google-cloud-python/bigquery/.nox/system-2-7/lib/python2.7/site-packages/IPython/external/decorators/__init__.py:2: DeprecationWarning: Importing from numpy.testing.decorators is deprecated since numpy 1.15.0, import from numpy.testing instead.
    from numpy.testing.decorators import *
.nox/system-2-7/lib/python2.7/site-packages/IPython/external/decorators/__init__.py:3
  /tmpfs/src/github/google-cloud-python/bigquery/.nox/system-2-7/lib/python2.7/site-packages/IPython/external/decorators/__init__.py:3: DeprecationWarning: Importing from numpy.testing.noseclasses is deprecated since 1.15.0, import from numpy.testing instead
    from numpy.testing.noseclasses import KnownFailure
tests/system.py::test_bigquery_magic
  /tmpfs/src/github/google-cloud-python/bigquery/.nox/system-2-7/lib/python2.7/site-packages/IPython/core/history.py:228: UserWarning: IPython History requires SQLite, your history will not be saved
    warn("IPython History requires SQLite, your history will not be saved")
-- Docs: https://docs.pytest.org/en/latest/warnings.html
1 failed, 49 passed, 3 warnings in 271.39 seconds
Command py.test --quiet tests/system.py failed with exit code 1
Session system-2.7 failed.
```

File: bigquery/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+docs/_build
```
Comment (busunkim96) [inline bigquery/.gitignore]: I think this is already being ignored in the top-level .gitignore?  https://github.com/googleapis/google-cloud-python/blob/cb1382e1b3bb606a8d527d988ccadec1ba9c29ec/.gitignore#L52

Comment (tswast) [reply bigquery/.gitignore]: No it wasn't, (git kept wanting to add the folder before I added the gitignore) but I can add it there to the top-level if you'd prefer?

Comment (busunkim96) [reply bigquery/.gitignore]: We'll be splitting the repo eventually anyway.  It's fine to keep it here. :+1:

**Pull request merged**